enterprise/tools/check_metrics/check_metrics.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"math"
+	"os"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil"
+	"github.com/buildbuddy-io/buildbuddy/server/util/flagutil/yaml"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/prometheus/client_golang/api"
+	"github.com/prometheus/common/model"
+	"golang.org/x/sync/errgroup"
+
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+const (
+	// This exit code is returned if the metrics look unhealthy and the rollout should proceed
+	metricsHealthyExitCode = 0
+)
+
+var (
+	config     = flagutil.New("spec", Config{}, "Config to specify which metrics should be monitored by this script.")
+	configPath = flag.String("config_path", "config.yaml", "Path to config file.")
+)
+
+// This script is intended to compare system health for a canary that has been deployed to the rest of the apps
+// If the metrics are healthy, this script will return exit code 0 indicating that the rollout should proceed
+// If the metrics are not healthy for the specified amount of time, the script will return an exit code indicating
+// that the release should be rolled back
+//
+// Pre-requisites:
+//   - You must create a tunnel to prometheus and port-forward in order to run this script with
+//     `kubectl --context="" --namespace=<NAMESPACE> port-forward "<POD_NAME>" --address 0.0.0.0 <FROM_PORT>:<TO_PORT>`
+func main() {
+	flag.Parse()
+	err := yaml.PopulateFlagsFromFile(*configPath)
+	if err != nil {
+		log.Fatalf("Could not read metrics config from %s: %s", *configPath, err)
+	}
+	err = log.Configure()
+	if err != nil {
+		log.Fatalf("Could not configure logger: %s", err)
+	}
+	prometheusClient, err := api.NewClient(api.Config{
+		Address: config.PrometheusAddress,
+	})
+	if err != nil {
+		log.Fatalf("Error creating prometheus client: %s\nDid you run the prometheus port-forward command?", err)
+	}
+	promAPI := v1.NewAPI(prometheusClient)
+
+	eg, gctx := errgroup.WithContext(context.Background())
+	for _, metric := range config.PrometheusMetrics {
+		metric := metric
+		eg.Go(func() error {
+			monitoringTimer := time.After(time.Duration(config.MonitoringTimeframeSeconds) * time.Second)
+			monitoringTicker := time.NewTicker(time.Duration(metric.PollingIntervalSeconds) * time.Second)
+			defer monitoringTicker.Stop()
+
+			unhealthyCount := 0
+			for {
+				select {
+				case <-gctx.Done():
+					return nil
+				case <-monitoringTimer:
+					return nil
+				case <-monitoringTicker.C:
+					healthy, err := metricHealthy(promAPI, metric.Name, metric.CanaryQuery, metric.BaselineQuery, metric.HealthThreshold)
+					if err != nil {
+						log.Warningf("Error querying metric %s: %s", metric.Name, err)
+					}
+
+					if healthy {
+						unhealthyCount = 0
+					} else {
+						unhealthyCount++
+					}
+
+					if unhealthyCount >= metric.MaxUnhealthyCount {
```

Comment:
Personally representing it as a count is more intuitive to me than a percentage. I don't necessarily tie the two values together. For example, if the liveness check is down for 5 minutes, that is a clear signal to me we need a rollback because 5 minutes is a long time for an app to be down, not because that's 50% of the total duration I want to monitor the metrics. 

If it sounds good to you we can leave it as is for now and refactor if people are finding it confusing to configure?

Reply:
Actually would it be more clear if it was MaxUnhealthyTimeframeSeconds instead of MaxUnhealthyCount?